### PR TITLE
03 more

### PR DESCRIPTION
**Pull request by igorchupin**
One more idea

### CODE REVIEW

File: store/src/main/java/ProductVariable.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+public class ProductVariable {
+    private static int tempPublisher = 101;
+    private final String publisherID;
+    private final int exemplarsLeft;
+    private final double price;
+    private final String variableID;
+    CoversTypes coverTypes;
+    private String publisher;
+    private String coverType;
+    private String coverID;
```
Comment (shukal94) [inline store/src/main/java/ProductVariable.java:16]: remove duplicates

File: store/src/main/java/ProductVariable.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        switch (types) {
+            case Hard:
+                coverID = "1";
+                break;
+            case Soft:
+                coverID = "2";
+                break;
+            case Exclusive:
+                coverID = "3";
+        }
```
Comment (shukal94) [inline store/src/main/java/ProductVariable.java:32]: useless

File: store/src/main/java/RandomStorePopulator.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            left = faker.number().numberBetween(1, 100);
+            ProductVariable variable3 = new ProductVariable(publisher, CoversTypes.Hard, priceProduct, left);
+            productVariables.add(variable3);
+        }
+        return productVariables;
+    }
+}
+
+
+
```
Comment (shukal94) [inline store/src/main/java/RandomStorePopulator.java:51]: remove blank lines

**Review by shukal94 — CHANGES_REQUESTED** [review]

File: store/src/main/java/ProductVariable.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    Hard,
+    Soft,
+    Exclusive
+}
+
+public class ProductVariable {
+    private String publisher;
+    private final int exemplarsLeft;
+    private final double price;
+    CoversTypes coverTypes;
```
Comment (shukal94) [inline store/src/main/java/ProductVariable.java:11]: coverType

File: store/src/main/java/ProductVariable.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
+enum CoversTypes {
+    Hard,
+    Soft,
+    Exclusive
+}
+
+public class ProductVariable {
+    private String publisher;
+    private final int exemplarsLeft;
```
Comment (shukal94) [inline store/src/main/java/ProductVariable.java:9]: quantity

File: store/src/main/java/ProductVariable.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    private final double price;
```
Comment (shukal94) [inline store/src/main/java/ProductVariable.java:10]: use BigDecimal for money

File: store/src/main/java/ProductVariable.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+    public String getPublisher() {
+        return publisher;
+    }
+
+    public void setPublisher(String publisher) {
+        this.publisher = publisher;
+    }
+
+    public String printVariable() {
```
Comment (shukal94) [inline store/src/main/java/ProductVariable.java:28]: this is toString responsibility, please rename that

**Review by shukal94 — CHANGES_REQUESTED** [review]

File: store/src/main/java/ProductVariable.java (hunk 1 above)
Comment (igorchupin) [reply store/src/main/java/ProductVariable.java]: Removed all about ID

File: store/src/main/java/ProductVariable.java (hunk 2 above)
Comment (igorchupin) [reply store/src/main/java/ProductVariable.java]: Removed

File: store/src/main/java/RandomStorePopulator.java (hunk 3 above)
Comment (igorchupin) [reply store/src/main/java/RandomStorePopulator.java]: Removed

File: store/src/main/java/ProductVariable.java (hunk 4 above)
Comment (igorchupin) [reply store/src/main/java/ProductVariable.java:11]: Changed

File: store/src/main/java/ProductVariable.java (hunk 5 above)
Comment (igorchupin) [reply store/src/main/java/ProductVariable.java:9]: Changed

File: store/src/main/java/ProductVariable.java (hunk 6 above)
Comment (igorchupin) [reply store/src/main/java/ProductVariable.java:10]: Changed

File: store/src/main/java/ProductVariable.java (hunk 7 above)
Comment (igorchupin) [reply store/src/main/java/ProductVariable.java:28]: Changed

**Review by shukal94 — APPROVED** [review]

**Pull request merged**